app/controllers/DisassociateDucrConfirmationController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.AppConfig
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.util.CacheIdGenerator.movementCacheId
+import forms.DisassociateDucr
+import forms.DisassociateDucr._
+import handlers.ErrorHandler
+import javax.inject.{Inject, Singleton}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.CustomsCacheService
+import uk.gov.hmrc.play.bootstrap.controller.FrontendController
+import views.html.disassociate_ducr_confirmation
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class DisassociateDucrConfirmationController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  customsCacheService: CustomsCacheService,
+  errorHandler: ErrorHandler,
+  mcc: MessagesControllerComponents,
+  disassociateDucrConfirmationPage: disassociate_ducr_confirmation
+)(implicit appConfig: AppConfig, ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayPage(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    customsCacheService
+      .fetchAndGetEntry[DisassociateDucr](movementCacheId, formId)
+      .map {
+        case Some(data) => Ok(disassociateDucrConfirmationPage(data.ducr))
+        case _ => Redirect(routes.StartController.displayStartPage())
```

Comment:
Is this correct? Or would we usually throw an exception when we require previously submitted info?

Reply:
Also should this cache entry be removed at any point?

Reply:
Perhaps it is start page as a missing data suggests an authenticated url hack?

Reply:
I don't think you need a cache in this case. There is just a single page, so you could take the data and send it immediately without storing in FE.
In declarations confirmation page we used Flash to pass data required.